### PR TITLE
Pull to refresh on profile

### PR DESCRIPTION
**Pull request by bfollington**
Fixes https://github.com/subconsciousnetwork/subconscious/issues/598
Fixes https://github.com/subconsciousnetwork/subconscious/issues/600

Reworked the structure to add empty states and make everything support `.refreshable{}`

https://github.com/subconsciousnetwork/subconscious/assets/5009316/b9f1ff2f-039e-4eb9-bfb9-cb592246aa13

### CODE REVIEW

File: xcode/Subconscious/Shared/Components/Common/Profile/UserProfileView.swift [hunk 1]
```diff
@@ -33,7 +33,7 @@ struct UserProfileView: View {
     var body: some View {
         let columnRecent = TabbedColumnItem(
```
Comment (gordonbrander) [inline xcode/Subconscious/Shared/Components/Common/Profile/UserProfileView.swift:34]: Nit: I tend to move any variables into private instance variables. It used to be the case that the view builder would get tripped up on the combination of statements and expressions.

**Review by gordonbrander — APPROVED** [review]
LGTM

**Comment by gordonbrander** [comment]
For future: `.refreshable` now takes an async function. The function will keep the spinner going until the async call completes.

We could consider a special exception to the way we typically handle effects:

```swift
.refreshable {
    // Any setup state
    store.send(.refreshing)
    let value = await store.environment.refreshThing()
    // Pass back into model update logic
    store.send(.refreshed(value))
}
```

**Comment by bfollington** [comment]
@gordonbrander if we had an API to wait for an action on the store we could keep it all in the model?

```swift
.refreshable {
    store.send(.refresh)
    await store.listen(action: .refreshed)
}
```

**Pull request merged**